hip-0000-registry-contract-for-naming-services.md

```diff
@@ -0,0 +1,160 @@
+Proposed Hedera centric registry contract that would also be extended to other inter-operable chains in the future would be able to accommodate multiple parties. Same registry contract would contain the real truth of mappings between Account IDs on the Mainnet & the corresponding human readable user-names. For example, john.hbar tied to 0.0.XXXXXXX
+
+## Specification
+
+Architecture of the proposed registry contract over Hedera would be, as follows:
+
+![RegistryContract](https://user-images.githubusercontent.com/97507177/181780608-e6bc217b-ee21-4b6d-a9a4-a28916a5ec87.png)
+
+ 
+(Image source: https://drive.google.com/file/d/1e3yxtaY8glEiptSXlrNxznxfzTfk1SKr/view) 
+
+## Detailed note on the flow
+
+**1. What is Registry Contract Suite?**<br />
+Registry Contract Suite is a group of smart Contracts developed over Solidity. These smart contracts are intended to register HTLDs (Hedera Top-Level-Domains) and Second-Level-Domains (SLDs) in a hierarchical manner. Domains and HTLDs once registered could be queried and looked up to avoid duplication.<br />
+**2. How does Registry Contract Suite Works?**<br />
+Registry Contract Suite is an intelligent group of smart contract that maintains an alphabetical hierarchy to register HTLDs and is smart enough to create and store HTLDs in their respective Alphabetical Registry Component.<br />
+	Once a HTLD is registered by the provider, an entry in respective alphabetical hierarchy smart contract is made. All further domains and subdomains under that hierarchy would be stored in that structure. So that for future, queries for that registered domain names or HTLD/s could be appropriately handled.<br />
+**3. Who are providers and how could someone become a provider of domains/HTLDs?**<br />
+Providers for Registry Contracts are the account addresses of Mainnet of entities who have access to register and update HTLD/s and domains. These providers could do a handful of operations and are provisioned to register a domain/HTLD. <br />
+To become a provider, entities need to raise a request to the admin of Registry Contracts, who have the privilege to add addresses as providers for the Registry Smart Contract Suite. Initially this would be done through a form accessible via the website of Web23 at web23.io with a goal of moving to a fully decentralized approach.<br />
+**4. Structure of Smart Contracts**<br />
+The structure of smart contracts are as follows:
+
+a.	**AlphaInterface:** This solidity file is an interface and it enables functions and abstract logic for the AToZRegistry Solidity Smart Contract. It enables functionalities such as register HTLD, activate TLD, etc.
+
+b.	**AlphaRegistry:** This solidity file is responsible for doing all the jobs and is exposed for providers. It would internally call different smart contracts from the suite to get the job done. 
+
+c.	**UTools:** This smart contract empowers the suite with a handful of utility functions like substring and all.
```

Comment:
string manipulation methods shouldn't be on chain, they aren't gas friendly and would cause undo burden on the network, at least for the near and mid future.